traitsui/handler.py

```diff
@@ -240,6 +242,80 @@ def show_help(self, info, control=None):
         on_help_call()(info, control)
 
     #-------------------------------------------------------------------------
+    #  Dispatches generic menu and toolbar events:
+    #-------------------------------------------------------------------------
+
+    def perform(self, info, action, event):
+        """ Perform computation for an action.
+
+        The default method looks for a method matching ``action.action`` and
+        calls it (sniffing the signature to determine how to call it for
+        historical reasons).  If this is not found, then it calls the
+        :py:meth:`perform` method of the action.
+
+        Parameters
+        ----------
+        info : UIInfo instance
+            The UIInfo assicated with the view, if available.
+        action : Action instance
+            The Action that the user invoked.
+        event : ActionEvent instance
+            The ActionEvent associated with the user action.
+
+        Notes
+        -----
+        If overriding in a subclass, the method needs to ensure that any
+        standard menu action items that are needed (eg. "Close", "Undo",
+        "Redo", "Help", etc.) get dispatched correctly.
+        """
+        if action.action != '':
+            method_name = action.action
+        else:
+            method_name = '_{}_clicked'.format(action.name.lower())
+
+        for object in self.get_perform_handlers(info):
+            method = getattr(object, method_name, None)
+            if method is not None:
+                # call the action method
+                specification = getargspec(method)
+                if len(specification.args) == 1:
+                    method()
+                else:
+                    method(info)
+                # and we are done
+                return
+
+        # otherwise, call the perform method of the action
+        specification = getargspec(action.perform)
+        if len(specification.args) == 1:
+            action.perform()
+        else:
+            action.perform(event)
+
+    def get_perform_handlers(self, info):
+        """ Return a list of objects which can handle actions.
+
+        This method may be overridden by sub-classes to return a more relevant
+        set of objects.
+
+        Parameters
+        ----------
+        info : UIInfo instance or None
+            The UIInfo associated with the view, or None.
+
+        Returns
+        -------
+        handlers : list
+            A list of objects that may potentially have action methods on them.
+        """
+        handlers = [self]
+        if info is not None:
+            additional_objects = ['object', 'model']
```

Comment:
The main reason is that due to the global module variable being remove, legacy code will have to subclasss in order to support the original behavior instead of setting a trait value